neverwinter/nwscript/native/scriptcompparsetree.cpp

```diff
@@ -524,6 +524,59 @@ int32_t CScriptCompiler::GenerateParseTree()
+                    pNewNode->m_psStringData = new CExoString(m_sCurrentFunction.CStr());
+                    pNewNode->m_bAllowAsDefaultValueInFunctionDecl = false;
+                    ModifySRStackReturnTree(pNewNode);
+                    return 0;
+                }
+                else if (m_nTokenStatus == CSCRIPTCOMPILER_TOKEN_KEYWORD_DASHDASH_FILE)
+                {
+                    CScriptParseTreeNode *pNewNode = CreateScriptParseTreeNode(CSCRIPTCOMPILER_OPERATION_CONSTANT_STRING,NULL,NULL);
+                    pNewNode->m_psStringData = new CExoString(m_pcIncludeFileStack[m_nCompileFileLevel-1].m_sCompiledScriptName.CStr());
+                    pNewNode->m_bAllowAsDefaultValueInFunctionDecl = false;
+                    ModifySRStackReturnTree(pNewNode);
+                    return 0;
+                }
+                else if (m_nTokenStatus == CSCRIPTCOMPILER_TOKEN_KEYWORD_DASHDASH_LINE)
+                {
+                    CScriptParseTreeNode *pNewNode = CreateScriptParseTreeNode(CSCRIPTCOMPILER_OPERATION_CONSTANT_INTEGER,NULL,NULL);
+                    pNewNode->m_bAllowAsDefaultValueInFunctionDecl = false;
+                    pNewNode->nIntegerData = m_nLines;
+                    ModifySRStackReturnTree(pNewNode);
+                    return 0;
+                }
+                else if (m_nTokenStatus == CSCRIPTCOMPILER_TOKEN_KEYWORD_DASHDASH_DATE)
+                {
+                    CScriptParseTreeNode *pNewNode = CreateScriptParseTreeNode(CSCRIPTCOMPILER_OPERATION_CONSTANT_STRING,NULL,NULL);
+                    char buffer[32];
+                    time_t now = time(0);
+                    struct tm tstruct = *localtime(&now);
+                    strftime(buffer, sizeof(buffer), "%Y-%m-%d", &tstruct);
+                    pNewNode->m_psStringData = new CExoString(buffer);
+                    ModifySRStackReturnTree(pNewNode);
+                    return 0;
+                }
+                else if (m_nTokenStatus == CSCRIPTCOMPILER_TOKEN_KEYWORD_DASHDASH_TIME)
+                {
+                    CScriptParseTreeNode *pNewNode = CreateScriptParseTreeNode(CSCRIPTCOMPILER_OPERATION_CONSTANT_STRING,NULL,NULL);
+                    char buffer[32];
+                    time_t now = time(0);
+                    struct tm tstruct = *localtime(&now);
+                    strftime(buffer, sizeof(buffer), "%H:%M:%S", &tstruct);
+                    pNewNode->m_psStringData = new CExoString(buffer);
+                    ModifySRStackReturnTree(pNewNode);
+                    return 0;
+                }
+                else if (m_nTokenStatus == CSCRIPTCOMPILER_TOKEN_KEYWORD_DASHDASH_SCRIPTCOMP_VERSION)
+                {
+                    CScriptParseTreeNode *pNewNode = CreateScriptParseTreeNode(CSCRIPTCOMPILER_OPERATION_CONSTANT_STRING,NULL,NULL);
+                    pNewNode->m_psStringData = new CExoString(SCRIPTCOMP_VERSION);
```

Comment:
nit: put this around an `#ifdef SCRIPTCOMP_VERSION` so it doesn't crash the game if pulled in. Game should probably fill it with the actual build version.

Reply:
IMO the game should also fill in the neverwinter.nim version it was based on, otherwise the value is kind of useless. 

Currently it's the nimble spec version, e.g. `1.8.1`, and it should remain that way.

Reply:
So when porting this over to the game, this value should probably be maintained and ported alongside. Don't think the ifdef is a good idea, it'll just mean it'll be missed.

Reply:
There should be a way to specify if the compiler was nw.nim, game or toolset. Whether they report a different string here, or if this is "scriptcomp lib version" doesn't matter much to me. But if the latter, then it shouldn't be tied to other nw.nim changes. Ie the value shouldn't update because of a bugfix in nwn_gff

Reply:
That makes sense to me, and thus the macro has the wrong name anyways. Can't deal with remaking the design here, I'll just yank it for now; if someone wants it they can add it back in. OK?

Reply:
Let's yank and we can add later after we're happy with it. I doubt anyone would use it anyway.